### PR TITLE
Upstream messages support

### PR DESCRIPTION
**Pull request by falsecz**
https://firebase.google.com/docs/cloud-messaging/android/upstream

SenderId can be obtained from google-services.json, but I don't know how to get `R.string.gcm_defaultSenderId` in submodule

### CODE REVIEW

**Comment by evollu** [comment]
you can convert readable map to a bundle. react native provides a utility

**Comment by falsecz** [comment]
- how can I pass bundle to RemoteMessage.Builder ?
- SenderID cannot be random: `A unique numerical value created when you create your Firebase project` https://firebase.google.com/docs/cloud-messaging/concept-options#senderid

**Comment by evollu** [comment]
does builder has addData or addExtras?
I feel the requirement is just for server sending message down to devices, not  the other way. Haven't tested though. In worst case you defined the ID as constant in your JS code and pass it through the function.

**Comment by falsecz** [comment]
- builder has only `addData(String key, String value)` and `setData(Map<String, String> data)` https://firebase.google.com/docs/reference/android/com/google/firebase/messaging/RemoteMessage.Builder
- I'm already passing senderId through the function: https://github.com/evollu/react-native-fcm/pull/132/files#diff-168726dbe96b3ce427e7fedce31bb0bcR77

File: android/src/main/java/com/evollu/react/fcm/FIRMessagingModule.java [hunk 1]
```diff
@@ -126,6 +128,38 @@ public void onReceive(Context context, Intent intent) {
...
+                return map.getString(key);
+            case Number:
+                try {
+                    return String.valueOf(map.getInt(key));
+                } catch (Exception e) {
+                    return String.valueOf(map.getDouble(key));
+                }
+            case Boolean:
+                return String.valueOf(map.getBoolean(key));
+            default:
```
Comment (evollu) [inline android/src/main/java/com/evollu/react/fcm/FIRMessagingModule.java]: what happens to nested data?

**Comment by evollu** [comment]
looks good

File: android/src/main/java/com/evollu/react/fcm/FIRMessagingModule.java (hunk 1 above)
Comment (falsecz) [inline android/src/main/java/com/evollu/react/fcm/FIRMessagingModule.java]: not supported, message builder has only Map<String, String>

Comment (evollu) [inline android/src/main/java/com/evollu/react/fcm/FIRMessagingModule.java]: so basic...

**Comment by evollu** [comment]
@falsecz do you have plan for adding iOS support?

**Pull request merged**